packages/nimble-components/src/theme-provider/tests/theme-provider.spec.ts

```diff
@@ -16,6 +19,79 @@ describe('Theme Provider', () => {
         );
     });
 
+    describe('lang token', () => {
+        async function setup(
+            langValue = 'foo'
+        ): Promise<Fixture<ThemeProvider>> {
+            return fixture<ThemeProvider>(
+                html`<${themeProviderTag} lang="${langValue}">
+                    </${themeProviderTag}>`
+            );
+        }
+
+        let element: ThemeProvider;
+        let connect: () => Promise<void>;
+        let disconnect: () => Promise<void>;
+
+        afterEach(async () => {
+            await disconnect();
+        });
+
+        it('value is set to "foo" when theme provider lang attribute is assigned value "foo"', async () => {
+            ({ element, connect, disconnect } = await setup());
+            await connect();
+            await waitForUpdatesAsync();
+            expect(lang.getValueFor(element)).toBe('foo');
+        });
+
+        it('value defaults to page lang when theme provider lang attribute is removed', async () => {
+            document.documentElement.lang = 'bar';
+            ({ element, connect, disconnect } = await setup());
+            await connect();
+            element.removeAttribute('lang');
+            await waitForUpdatesAsync();
+            expect(lang.getValueFor(element)).toBe('bar');
+        });
+
+        it('value defaults to page lang when theme provider lang attribute is empty string', async () => {
+            document.documentElement.lang = 'bar';
+            ({ element, connect, disconnect } = await setup(''));
+            await connect();
+            await waitForUpdatesAsync();
+            expect(lang.getValueFor(element)).toBe('bar');
+            expect(element.validity.invalidLang).toBeFalse();
+        });
+
+        it('value defaults to page lang when theme provider lang attribute is malformed', async () => {
+            document.documentElement.lang = 'bar';
+            ({ element, connect, disconnect } = await setup('123'));
+            await connect();
+            await waitForUpdatesAsync();
+            expect(lang.getValueFor(element)).toBe('bar');
+            expect(element.validity.invalidLang).toBeTrue();
+        });
+
+        it('value updates when theme provider lang attribute goes from bad to good', async () => {
```

Comment:
I think one more test here would be good that does this opposite of this (verifies that the correct thing happens when the theme provider's lang attribute is valid and then gets set to something invalid).